src/persist/src/indexed/columnar/parquet.rs

```diff
@@ -10,64 +10,106 @@
 //! Apache Parquet encodings and utils for persist data
 
 use std::io::{Read, Seek, Write};
+use std::sync::Arc;
 
 use arrow2::io::parquet::read::{infer_schema, read_metadata, FileReader};
 use arrow2::io::parquet::write::{
     CompressionOptions, Encoding, FileWriter, KeyValue, RowGroupIterator, Version, WriteOptions,
 };
 use differential_dataflow::trace::Description;
+use mz_dyncfg::Config;
+use mz_ore::bytes::SegmentedBytes;
 use mz_persist_types::Codec64;
 use timely::progress::{Antichain, Timestamp};
 
 use crate::error::Error;
 use crate::gen::persist::ProtoBatchFormat;
 use crate::indexed::columnar::arrow::{
-    decode_arrow_batch_kvtd, encode_arrow_batch_kvtd, SCHEMA_ARROW_KVTD,
+    decode_arrow_batch_kvtd, decode_arrow_batch_kvtd_arrow_rs, encode_arrow_batch_kvtd,
+    encode_arrow_batch_kvtd_arrow_rs, SCHEMA_ARROW2_KVTD, SCHEMA_ARROW_RS_KVTD,
 };
 use crate::indexed::columnar::ColumnarRecords;
 use crate::indexed::encoding::{
     decode_trace_inline_meta, encode_trace_inline_meta, BlobTraceBatchPart,
 };
 use crate::metrics::ColumnarMetrics;
 
+pub(crate) const USE_ARROW_RS_LIBRARY: Config<bool> = Config::new(
+    "persist_use_arrow_rs_library",
```

Comment:
Oh, sorry, one last thing! I'm personally inclined to split this into two dyncfgs, one for the read path and one for the write path. Worse for the new checks test, but better for the rollout (e.g. can turn on reads before writes). Thoughts?

Reply:
what do you think about one dyncfg with three values: 

1. "off"
2. "read"
3. "read_and_write"

I think being able to turn on reads before writes is great! But I wouldn't want to turn on writes and not reads. What do you think, maybe it's too much?

Reply:
My understanding is it should be perfectly safe to toggle the reads and writes independently. If that's correct, then I have a weak preference for the two bools (more directly reflects the optionality we have, also less Config boilerplate), but I'd also accept the three value enum if you feel strongly!

Reply:
Actually, scratch that. I think I'm probably close enough to 50/50 on two bools vs three variant that it's a wash. Your call!

Reply:
Opted for the one dyncfg! It's easier to test and while we totally should be able to support writing with `arrow-rs` and reading with `arrow2` we shouldn't ever need to. And I'm happy to cut scope wherever possible :)